src/Hosting/TestHost/src/ResponseBodyReaderStream.cs

```diff
@@ -124,8 +124,8 @@ internal void Cancel()
         internal void Abort(Exception innerException)
         {
             Contract.Requires(innerException != null);
-            _aborted = true;
             _abortException = innerException;
+            _aborted = true;
```

Comment:
nit: why change this?

Reply:
I put a for loop of 1,000,000 around the test logic to see if the error went away. There was a race condition about the order the properties are set that comes up here:

```cs
private void CheckAborted()
{
    if (_aborted)
    {
        throw new IOException(string.Empty, _abortException);
    }
}
```

You could get an IOException with a null inner exception.

Reply:
This kind of pattern scares me. Most people expect to set a bool like `_aborted` at the at the top of a method called Aborted. Would it be better just to lock when checking aborted and setting the aborted Exception?

Reply:
Is locking necessary here? We just need to put the class in the right state before setting _aborted to true.

I could add a comment to Aborted if you'd like.

Reply:
eh, I think it's fine. Not worth rerunning CI for.

Reply:
If something is concurrently accessing this state then it should be locked.

Reply:
This?

```cs
        private readonly object _abortLock = new object();

        internal void Abort(Exception innerException)
        {
            Contract.Requires(innerException != null);

            lock (_abortLock)
            {
                _abortException = innerException;
                _aborted = true;
                _pipe.Reader.CancelPendingRead();
            }
        }

        private void CheckAborted()
        {
            lock (_abortLock)
            {
                if (_aborted)
                {
                    throw new IOException(string.Empty, _abortException);
                }
            }
        }
```